src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -687,6 +687,34 @@ public boolean setPassenger(@NotNull Entity passenger)
 		return Collections.unmodifiableList(this.passengers);
 	}
 
+	/**
+	 * Gets a list of passengers on this vehicle.
+	 *
+	 * @param transitive Whether transitive passengers should be included (passengers of passengers).
+	 * @return An immutable list of passengers.
+	 */
+	public @NotNull List<Entity> getPassengers(boolean transitive)
+	{
+		if (!transitive)
+		{
+			return getPassengers();
+		}
+		List<Entity> entities = new ArrayList<>();
+		for (Entity passenger : this.passengers)
+		{
+			entities.add(passenger);
+			if (passenger instanceof EntityMock mock)
+			{
+				entities.addAll(mock.getPassengers(true));
+			}
+			else
+			{
+				entities.addAll(passenger.getPassengers());
+			}
```

Comment:
```suggestion
			entities.addAll(((EntityMock) passenger).getPassengers(transitive));
```
Passengers are [always instances of `EntityMock`](https://github.com/MockBukkit/MockBukkit/blob/e5631cc1b5da36af1cbfd7d04ee2412fcd55619c/src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java#L713) in order to have a reference to the vehicle in the passenger.

Using `transitive` is more descriptive than using `true`. Most IDEs would warn that this could be simplified by using `true` directly, and that's the case. Conceptually, the `transitive` parameter completely changes the algorithm used, so in my opinion, a new method with another name (like the first implementation in this PR) would be preferable as it is quickly identifiable.

Reply:
I think leaving it as an overload is a little simpler, but I can definitely see where confusion could come from. @thelooter What do you think?

Reply:
I personally prefer the version  @syldium proposed